lesson_10/libraries/src/loaders/lj_mcwilliams_loader.ts

```diff
@@ -0,0 +1,57 @@
+import csv from 'csv-parser';
+import fs from 'fs';
+import { Credit, MediaItem } from '../models/index.js';
+import { Loader } from './loader.js';
+
+export class LjMcwilliamsLoader implements Loader {
+  getLoaderName(): string {
+    return 'ljmcwilliams';
+  }
+
+  async loadData(): Promise<MediaItem[]> {
+    const credits = await this.loadCredits();
+    const mediaItems = await this.loadMediaItems();
+
+    console.log(
+      `Loaded ${credits.length} credits and ${mediaItems.length} media items`,
+    );
+
+    return [...mediaItems.values()];
+  }
+
+  async loadMediaItems(): Promise<MediaItem[]> {
+    // TODO: Implement this method.
+    const media = [];
+    const readable = fs
+      .createReadStream('data/media_items.csv', 'utf-8')
+      .pipe(csv());
+    for await (const row of readable) {
+      /**this destructures the CSV file rows */
+      const { id, type, title, year } = row;
+      media.push(new MediaItem(id, title, type, year, []));
+    }
+    return media;
+  }
+
+  /*
+  an asyncchronous function named loadCredits
```

Comment:
Capitalize the beginning of sentences and add proper punctuation at the end, here and below.

Reply:
Just made these changes. Working on the second request.